ADRs/RH/SIG-SRE/ADR-00007-Error-Budget-Policy.md

```diff
@@ -0,0 +1,250 @@
+
+For an SLO to be effective, it needs to have consequences. That is:
+
+"What happens when we consume all our error budget?"
+
+In many organisations, this question is often answered through the existence of a document, backed by a process, that can be called an "Error Budget Policy".
+
+
+## Goals
+
+The aim of this ADR is to provide a template, with some rationale, for an example Error Budget Policy.
+
+
+## Non-goals
+
+* This ADR does not aim at providing a definitive policy. The intention would be that consumers of this document would be able to take this example, based on the lessons learned from SRE teams already experienced in the running of services, and mold it to their own experience. It is purposefully generalised.
+
+
+## Current Architecture
+
+* An analysis of a number of existing Error Budget Policies was conducted, identifying the commonalities and the differences. The best practices articulated in this document was a synthesis of these sources.
+
+
+## Proposed Architecture
+
+The proposed content is as follows:
+
+***
+
+# Error Budget Policy (draft)
+
+Service Name: &lt;Service/s Name&gt;
+
+# Service
+
+&lt;Briefly describe the service/s under the scope of this document.&gt;
+
+# Definitions
+
+* _SLI_: The Service Level Indicator is the carefully-considered measure for an aspect of the service that is considered necessary to its function (e.g. the login screen should be available).
+* _SLO_: The Service Level Objective is the desired target for the aspect measured by an SLI (e.g. the login screen should be available for 99% of the time).
+* _Error Budget_:  The Error budget is one of the fundamental metrics for service reliability. It is the measure of how much a service can be unreliable (e.g. fail) without impacting customers. Error budgets are always based on SLOs.
+* _SLO miss_: The state where the error budget has been consumed, and the service is not operating "as expected". 
+* _Reliability Work_: The work that is needed to recover the service from an "SLO miss" state.
+* _Feature Work_: The work that is done to provide the features and functionality that is needed by that service.
+
+# Aim
+
+There are many inputs which affect the prioritization of work. The reliability of a service is a significant, but not the only, input to this process. The Error Budget is the primary metric that is used to inform the decision makers of issues that affect the reliability of a service, and by inference, the satisfaction of the customer with that service.
+
+ The question it is aiming to help answer is:
+
+"What do we need to do now that the error budget is all gone?"
+
```

Comment:
I would also see the aim of an error budget policy as a way to detect SLO miss early by introducing Multiple Burn Rate Alerts. In this case, we would need to answer also the following question: "How do we detect the initial signs of system degradation, and what **automated** mechanisms do we have in place for an interception?"

Reply:
I have added this (slightly modified) version in the next commit:

The questions that this is aiming to help answer is:

* "What do we need to do now that the error budget is all gone?"
* "How do we detect the early signs of system degradation, and what automated mechanisms do we have in place to intercept them?"